src/docfx/lib/ProcessUtility.cs

```diff
@@ -16,46 +16,41 @@ namespace Microsoft.Docs.Build
     internal static class ProcessUtility
     {
         /// <summary>
-        /// Execute process with args
+        /// Start a new process and wait for its execution asynchroniously
         /// </summary>
-        /// <param name="fileName">The process path name or location</param>
-        /// <param name="commandLineArgs">The process command line args</param>
-        /// <param name="cwd">The current working directory</param>
-        /// <param name="timeout">The timeout setting, default is none</param>
-        /// <param name="outputHandler">The process output action</param>
-        /// <returns>The executed result</returns>
-        public static Task<string> Execute(string fileName, string commandLineArgs, string cwd = null, TimeSpan? timeout = null, Action<string, bool> outputHandler = null)
+        public static Task<string> Execute(string fileName, string commandLineArgs, string cwd = null, TimeSpan? timeout = null, bool redirectOutput = true)
         {
             Debug.Assert(!string.IsNullOrEmpty(fileName));
 
             var tcs = new TaskCompletionSource<string>();
 
             var error = new StringBuilder();
             var output = new StringBuilder();
+            var psi = new ProcessStartInfo
+            {
+                FileName = fileName,
+                WorkingDirectory = cwd,
+                Arguments = commandLineArgs,
+            };
+
+            if (redirectOutput)
+            {
+                psi.UseShellExecute = false;
+                psi.CreateNoWindow = true;
```

Comment:
>psi.CreateNoWindow = true [](start = 16, length = 25)

should we always not create new windows ?

Reply:
it should always set UseShellExecute = false. CreateNoWindow = false will suppress the output

---
In reply to: [201551877](https://github.com/dotnet/docfx/pull/3036#discussion_r201551877) [](ancestors = 201551877)